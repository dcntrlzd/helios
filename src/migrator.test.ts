import * as path from 'path';
import Migrator from './migrator';

const MIGRATIONS_DIRECTORY = path.resolve(__dirname, '../migrations');

it('can load a sorted list of migrations', () => {
  const migrator = new Migrator(MIGRATIONS_DIRECTORY);
  expect(migrator.getMigrations()).toMatchSnapshot();
});

it('runs the migrations in order', async () => {
  const migrator = new Migrator(MIGRATIONS_DIRECTORY);
  await migrator.run();
});
