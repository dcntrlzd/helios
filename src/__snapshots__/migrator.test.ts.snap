// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can load a sorted list of migrations 1`] = `
Array [
  Object {
    "id": 1,
    "name": "01-manager.js",
    "path": "/Users/matrushka/dev/dcntrlzd/helios/test/migrations/01-manager.js",
    "runner": [Function],
  },
  Object {
    "id": 2,
    "name": "02-deployContracts.js",
    "path": "/Users/matrushka/dev/dcntrlzd/helios/test/migrations/02-deployContracts.js",
    "runner": [Function],
  },
]
`;
